etcdserver/server.go

```diff
@@ -437,7 +437,7 @@ func (s *EtcdServer) ReportSnapshot(id uint64, status raft.SnapshotStatus) {
 }
 
 func (s *EtcdServer) run() {
-	snap, err := s.r.raftStorage.Snapshot()
+	snap, err := s.r.raftStorage.MemoryStorage.Snapshot()
```

Comment:
add a comment for this. Also we need to add a TODO to clean this up... Probably a new method name. It is not clear what is going on here by reading the code.